terraform/modules/department/50-aws-iam-roles.tf

```diff
@@ -41,55 +41,68 @@ data "aws_iam_policy_document" "glue_agent_assume_role" {
   }
 }
 
+# Permissions for the departmental glue agent
 resource "aws_iam_role" "glue_agent" {
   tags = var.tags
 
   name               = lower("${var.identifier_prefix}-glue-${local.department_identifier}")
   assume_role_policy = data.aws_iam_policy_document.glue_agent_assume_role.json
 }
 
-resource "aws_iam_role_policy_attachment" "glue_agent_s3_access" {
-  role       = aws_iam_role.glue_agent.name
-  policy_arn = aws_iam_policy.s3_access.arn
-}
-
-resource "aws_iam_role_policy_attachment" "glue_agents_secrets_manager_read_only" {
-  role       = aws_iam_role.glue_agent.name
-  policy_arn = aws_iam_policy.glue_can_write_to_cloudwatch.arn
-}
 
-resource "aws_iam_role_policy_attachment" "glue_agent_glue_scripts_read_only" {
-  role       = aws_iam_role.glue_agent.name
-  policy_arn = aws_iam_policy.glue_scripts_read_only.arn
+# Define a map for the policies
+locals {
+  policy_map = {
+    s3_access                                      = aws_iam_policy.s3_access.arn,
+    glue_can_write_to_cloudwatch                   = aws_iam_policy.glue_can_write_to_cloudwatch.arn,
+    glue_scripts_read_only                         = aws_iam_policy.glue_scripts_read_only.arn,
+    secrets_manager_read_only                      = aws_iam_policy.secrets_manager_read_only.arn,
+    full_glue_access                               = aws_iam_policy.full_glue_access.arn,
+    crawler_can_access_jdbc_connection             = aws_iam_policy.crawler_can_access_jdbc_connection.arn,
+    full_s3_access_to_glue_resources               = aws_iam_policy.full_s3_access_to_glue_resources.arn,
+    glue_access_to_watermarks_table                = aws_iam_policy.glue_access_to_watermarks_table.arn,
+    glue_runner_pass_role_to_glue_for_notebook_use = aws_iam_policy.glue_runner_pass_role_to_glue_for_notebook_use.arn,
+  }
 }
 
-resource "aws_iam_role_policy_attachment" "glue_agent_glue_can_write_to_cloudwatch" {
+# Attach policies to the departmental Glue IAM role
+resource "aws_iam_role_policy_attachment" "glue_agent_policy_attachment" {
+  for_each   = local.policy_map
   role       = aws_iam_role.glue_agent.name
-  policy_arn = aws_iam_policy.secrets_manager_read_only.arn
+  policy_arn = each.value
 }
 
-resource "aws_iam_role_policy_attachment" "glue_agent_glue_full_access" {
-  role       = aws_iam_role.glue_agent.name
-  policy_arn = aws_iam_policy.full_glue_access.arn
+# IAM user and permission for departmetnal Airflow user
+resource "aws_iam_user" "airflow_user" {
+  count = var.departmental_airflow_user ? 1 : 0
+  name  = "${local.department_identifier}-airflow-user"
+  tags  = var.tags
 }
 
-resource "aws_iam_role_policy_attachment" "crawler_can_access_jdbc_connection" {
-  role       = aws_iam_role.glue_agent.name
-  policy_arn = aws_iam_policy.crawler_can_access_jdbc_connection.arn
+resource "aws_iam_user_policy_attachment" "airflow_user_policy_attachment" {
```

Comment:
I am thinking if we should differentiate the Airflow policy from the Glue policy.

If we define a base_airflow_policy, the airflow_policy_map in the locals block would only include:
```
base_airflow_policy 
secrets_manager_read_only # using existing one
s3_access # using existing one
```

How's your idea? @timburke-hackit . If it doesn't matter at current stage, then this PR is ready to be review and merge. Otherwise, I will update the PR shortly. Thank you.

Reply:
Yeah, I was thinking the same thing on Friday but was also trying to figure out some other issues which is why I haven't approved this yet but wanted to come back to it fresh this morning.

Reply:
Gotcha, @timburke-hackit Thanks for confirmation. I will update the PR now.